From ebc21aa17354437fd9a0f2d15e0d551ae825b958 Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Wed, 28 Jan 2015 16:06:00 -0800
Subject: [PATCH] REVERTME: unfatalize refcount errors during ISP resets

Under various poorly-understood conditions (this has happened
repeatedly on different devices and is believed to be sensitive to
external sensor timing) the ISP can hit a watchdog reset condition in
a state where the MMU refcounts are invalid.  These will panic the
kernel.

But since the only known case where this is hit is a reset of the
device, it is believed that making these a benign warning instead is
safe.

Change-Id: I9ce3f3b6045ec783d616a7db819de6a72d791575
Tracked-On: https://jira01.devtools.intel.com/browse/IRDA-3451
Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 .../pci/atomisp2/css2400b0_v21/base/refcount/src/refcount.c    | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/base/refcount/src/refcount.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/base/refcount/src/refcount.c
index 3e2e19d..1a554427 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/base/refcount/src/refcount.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/base/refcount/src/refcount.c
@@ -203,9 +203,10 @@ bool ia_css_refcount_decrement(int32_t id, hrt_vaddress ptr)
 			id, ptr, entry, entry->id, entry->count);
 	else
 		IA_CSS_ERROR("entry NULL\n");
-	assert(false);
 
-	return false;
+	WARN_ONCE(1, "CSS refcount failure entry %p id %d", entry, id);
+
+	return true;
 }
 
 bool ia_css_refcount_is_single(hrt_vaddress ptr)
@@ -253,10 +254,7 @@ void ia_css_refcount_clear(int32_t id, clear_func clear_func_ptr)
 						    "no clear_func\n");
 				mmgr_free(entry->data);
 			}
-			assert(entry->count == 0);
-			if (entry->count != 0) {
-				IA_CSS_WARNING("Ref count for entry %x is not zero!", entry->id);
-			}
+			WARN_ONCE(1, "CSS refcount failure entry %p id %d", entry, id);
 			entry->data = mmgr_NULL;
 			entry->count = 0;
 			entry->id = 0;
-- 
2.1.0

